src/obfuscation/hidden.h

```diff
@@ -13,35 +13,45 @@
 extern "C" {
 #endif
 
-typedef struct HiddenName HiddenName;
+typedef struct HiddenString HiddenString;
 
 // Hides the string, obfuscation is done elsewhere
 // Should discourage directly accessing the string and printing out user data
-HiddenName *NewHiddenName(const char *name, uint64_t length, bool takeOwnership);
-void HiddenName_Free(HiddenName *value, bool tookOwnership);
-
-// comparison
-int HiddenName_Compare(const HiddenName *left, const HiddenName *right);
-int HiddenName_CompareC(const HiddenName *left, const char *right, size_t right_length);
-int HiddenName_CaseInsensitiveCompare(HiddenName *left, HiddenName *right);
-int HiddenName_CaseInsensitiveCompareC(HiddenName *left, const char *right, size_t right_length);
-
-// ownership managment
-HiddenName *HiddenName_Duplicate(const HiddenName *value);
-void HiddenName_TakeOwnership(HiddenName *hidden);
-void HiddenName_Clone(HiddenName *src, HiddenName **dst);
-
-// allowed actions
-void HiddenName_SaveToRdb(HiddenName* value, RedisModuleIO* rdb);
-void HiddenName_DropFromKeySpace(RedisModuleCtx* redisCtx, const char* fmt, HiddenName* value);
-RedisModuleString *HiddenName_CreateString(HiddenName* value, RedisModuleCtx* ctx);
+// This is a security measure to prevent leaking user data
+// The additional takeOwnership determines whether to duplicate the buffer or directly point at the given buffer
+// HiddenString_Free must be called for the object do release it
+HiddenString *NewHiddenString(const char *name, size_t length, bool takeOwnership);
+// Frees a hidden string, if takeOwnership is true, the buffer is freed as well
+void HiddenString_Free(const HiddenString *value, bool tookOwnership);
```

Comment:
I agree the signature might be confusing - ownership wise
Note it changes in next PR to be a ref counted object, so the bool argument gets removed.

The const was needed because of recent changes adding const to some `HiddenString` members.
I preferred to avoid hard casts to remove the const in all the code and just have the free accept a const pointer.